fix: await getInitialData() in APP_INITIALIZER to prevent localization timing issues

fahrigedik opened the pull request:
### Description

Resolves https://github.com/volosoft/volo/issues/20421

TODO: 
Hi @sumeyyeKurtulus 
**problem :**
- [x] Components rendered without loading localisation data.
- [x] Timing Issue: components were rendering before getInitialData() completed
- [x] Race Condition: Localization data vs component rendering

**Solution :**  
- await getInitialData() - Waiting until localisation data is fully loaded
- Data is ready before components render

**Execution Flow :** 

_Broken_
```
App Start → APP_INITIALIZER → getInitialData() starts (no await) 
         → Components render → Localization not ready → Shows "Settings"
```

_Fixed :_ 
```
App Start → APP_INITIALIZER → await getInitialData() → Localization loaded
         → await localeInitializer() → Locale set → Components render
         → Localization ready → Shows "Ayarlar"
```


### Checklist

- [ ] I fully tested it as developer / designer and created unit / integration tests
- [ ] I documented it (or no need to document or I will create a separate documentation issue)

### How to test it?

**Step 0 :** 
You can use this command under `abp > npm > ng-packs`: `yarn ng g @abp/ng.schematics:change-theme --name 2 --target-project dev-app  
`

**Step 1 :** 
[project.json](https://github.com/user-attachments/files/21571234/message.6.txt)
 Replace `volo/abp/npm/ng-packs/apps/dev-app/project.json` with the above.
 
 
 **Step 2 :** 
Replace `volo/abp/npm/ng-packs/tsconfig.base.json` lines 23 and 24 with the following.
```
"@abp/ng.core": [
 "../../../../abp/npm/ng-packs/packages/core/src/public-api.ts",
 ],
 "@abp/ng.core/locale": [
 "../../../../abp/npm/ng-packs/packages/core/locale/src/public-api.ts",
],
````

**Step 3 :**
Remove `InternetConnectionStatusComponent`, from abp/npm/ng-packs/apps/dev-app/src/app/app.module.ts

**Step 4 :**  

In `volo/abp/npm/ng-packs/apps/dev-app/src/app/app.component.ts` delete 
```
 <abp-gdpr-cookie-consent />
 <abp-internet-status />
 <abp-loader-bar />
```
and remove their dependent imports.

**Step 5 :** 

`volo/abp/npm/ng-packs` yarn install and yarn run.
and 
`volo/abp/templates/app-pro/aspnet-core/src/MyCompanyName.MyProjectName.HttpApi.HostWithIds` on dotnet run

fahrigedik committed:
Refactor locale initialization to support async and injector

Updated provideAppInitializer to use an async function and pass the Injector to localeInitializer. Modified localeInitializer to accept an optional Injector parameter for improved flexibility and testability.

masum-ulu commented on npm/ng-packs/packages/core/src/lib/utils/initial-utils.ts:
```diff
@@ -52,10 +52,10 @@ export async function getInitialData() {
   await lastValueFrom(result$);
```
Call `localeInitializer` method at here and remove from core-module-config.provider.ts

fahrigedik committed:
Move localeInitializer call to getInitialData

The localeInitializer invocation was moved from the app initializer in core-module-config.provider.ts to the getInitialData function in initial-utils.ts. This change centralizes initialization logic and ensures locale setup occurs after initial data is loaded.

sumeyyeKurtulus commented on npm/ng-packs/packages/core/src/lib/providers/core-module-config.provider.ts:
```diff
@@ -107,13 +107,14 @@ export function provideAbpCore(...features: CoreFeature<CoreFeatureKind>[]) {
       inject(RoutesHandler);
-    }),
 
+      return (async (): Promise<void> => {
```
We could actually use such implementation instead:

```ts
    provideAppInitializer(async () => {
      inject(LocalizationService);
      inject(LocalStorageListenerService);
      inject(RoutesHandler);
      await getInitialData();
    })
```

 
The direct async approach is cleaner and more readable since it eliminates unnecessary function wrapping.

It also follows Angular conventions as this is the standard pattern for the old APP_INITIALIZER, and reduces complexity because `provideAppInitializer` handles async functions natively.

fahrigedik committed:
Refactor app initializer to use async function

Simplifies the app initializer by making it an async function directly instead of returning an async IIFE. This improves readability and maintains the same initialization behavior.

sumeyyeKurtulus committed:
update: remove unused imports

yagmurcelk reviewed (changes requested):
The localization issue that occurs when switching between certain languages still persists.

<img width="1898" height="824" alt="image" src="https://github.com/user-attachments/assets/ef92d85a-f27b-4dfd-889c-70ea68bb8f6d" />

fahrigedik committed:
Add Czech translations for Marked Items and comments

Added new Czech localization strings for Marked Items, favorites, and comment approval features in cs.json. This update supports new UI elements and system messages related to marking items, filtering favorites, and comment moderation.
